sdk/nodejs/runtime/closure/createClosure.ts

```diff
@@ -1017,16 +1024,22 @@ async function getOrCreateEntryAsync(
             // Now, make an entry just for this name.
             const keyEntry = await getOrCreateNameEntryAsync(propName, undefined, context, serialize, logInfo);
 
-            if (environment.has(keyEntry)) {
+            // We're about to recurse inside this object.  In order to prevent infinite loops, put a
+            // dummy entry in the environment map.  That way, if we hit this object again while
+            // recursing we won't try to generate this property.
+            //
+            // Note: we only stop recursing if we hit exactly our sentinel key (i.e. we're self
+            // recursive).  We *do* want to recurse through the object again if we see it through
+            // non-recursive paths.  That's because we might be hitting this object through one
+            // prop-name-path, but we created it the first time through another prop-name path.
+            //
+            // By processing the object again, we will add the different members we need.
+            if (object.env.has(keyEntry) && object.env.get(keyEntry) === undefined) {
                 continue;
             }
+            object.env.set(keyEntry, <any>undefined);
```

Comment:
same fix as above.  we prevent infinite recursion.  but we don't stop re-examination of props we've already seen.